### PR TITLE
[Container] `az container create`: Add new parameters for container security context for confidential ContainerGroupSku

### PR DESCRIPTION
**Pull request by fnuarnav**
**Related command**

`az container create `

**Description**

This PR:
upgrades azure-mgmt-containerinstance to 10.1.0
Adds Container security context properties for confidential ContainerGroupSku:
- privileged: run the container as a privileged container
- allow-privilege-escalation: allow containers to be run with elevated privileges
- run-as: run container process as a user
- run-as-group: run container process as a user group
- seccomp-profile: base64 encoded seccomp profile
- add-capabilities: list of capabilities to be added
- drop-capabilities: list of capabilities to be dropped

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Container

**Comment by fnuarnav** [comment]
I tried recording the test test_tag_update_by patch under command_modules/resource by running `azdev test test_tag_update_by_patch`, but the test was failing with error "CLIError: Please run az login to setup account",  even after running `az login`, and also when I tried removing the recording so that the test is run during CI tests.

image

I looked at the context and found that a dummy context is being passed to the method, which has SubscriptionId=None.

**Comment by zhoxing-ms** [comment]
@fnuarnav I cannot reproduce your issue locally. Can you try using `az login`  to login first, and then use `az account show` to check the subscription being used and then execute `azdev test test_tag_update_by_patch --live` to re-record the yaml file?

File: src/azure-cli/azure/cli/command_modules/container/_params.py [hunk 1]
```diff
@@ -79,6 +79,13 @@ def load_arguments(self, _):
 
     with self.argument_context('container create', arg_group='Confidential Container Group') as c:
         c.argument('cce_policy', help="The CCE policy for the confidential container group")
+        c.argument('allow_privilege_escalation', options_list=['--allow-escalation'], help="Allows whether a process can gain more privileges than its parent process.", action='store_true')
+        c.argument('privileged', help='The flag to determine if the contianer permissions is elevated to Privileged', action='store_true')
+        c.argument('run_as_user', help="Sets the User GID for the container")
+        c.argument('run_as_group', help="Sets the User UID for the container")
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/container/_params.py:85]: Please use the first-person voice in the help message
```suggestion
        c.argument('allow_privilege_escalation', options_list=['--allow-escalation'], help="Allow whether a process can gain more privileges than its parent process.", action='store_true')
        c.argument('privileged', help='The flag to determine if the contianer permissions is elevated to Privileged', action='store_true')
        c.argument('run_as_user', help="Set the User GID for the container")
        c.argument('run_as_group', help="Set the User UID for the container")
```

File: src/azure-cli/azure/cli/command_modules/container/_params.py [hunk 2]
```diff
@@ -79,6 +79,13 @@ def load_arguments(self, _):
...
+        c.argument('seccomp_profile', help="A base64 encoded string containing the contents of the JSON in the seccomp profile")
+        c.argument('add_capabilities', type=str, nargs='+', help="A List of security context capabilities to be added")
+        c.argument('drop_capabilities', type=str, nargs='+', help="A List of security context capabilities to be dropped")
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/container/_params.py:88]: `type=str` is the default setting, please do not explicitly repeat the declaration
```suggestion
        c.argument('add_capabilities', nargs='+', help="A List of security context capabilities to be added")
        c.argument('drop_capabilities', nargs='+', help="A List of security context capabilities to be dropped")
```

Comment (fnuarnav) [reply src/azure-cli/azure/cli/command_modules/container/_params.py]: > @fnuarnav I cannot reproduce your issue locally. Could you try using `az login` to login first, and then use `az account show` to check the subscription being used and then execute `azdev test test_tag_update_by_patch --live` to re-record the yaml file?

I have tried logging in and running the test but still gives the same error.

Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/container/_params.py]: @fnuarnav May I ask if the subscription output after executing `az login` meets your expectations?

Comment (fnuarnav) [reply src/azure-cli/azure/cli/command_modules/container/_params.py]: Yes, after `az login` the subscription output is what I would expect. I was able to run the tests under the container command_module with this subscription.

**Comment by zhoxing-ms** [comment]
@fnuarnav Could you please fix these CI issues?

**Comment by zhoxing-ms** [comment]
@fnuarnav Please note that we are launching the release for this sprint this week. Please resolve all comments by tomorrow, otherwise the release of this PR will have to be postponed to the next sprint (on **07-04**)

**Comment by zhoxing-ms** [comment]
> I tried recording the test test_tag_update_by patch under command_modules/resource by running azdev test test_tag_update_by_patch, but the test was failing with error "CLIError: Please run az login to setup account", even after running az login, and also when I tried removing the recording so that the test is run during CI tests.

I found that the issue with recording tests was caused by PR #25689
```python
class TagScenarioTest(ScenarioTest):

    def __init__(self, *arg, **kwargs):
        super().__init__(*arg, random_config_dir=True, **kwargs)
```
The random `config_dir` can result in the inability to find the corresponding subscription configuration when recording tests in live mode.

**Review by zhoxing-ms — APPROVED** [review]

**Review by yanzhudd — APPROVED** [review]

**Comment by zhoxing-ms** [comment]
> how did you work around the random_config_dir=True issue, by temporarily removing it?

@jiasli Yes, since it is not related to the business logic we are testing, so I removed it temporarily to quickly unblock our recording

**Pull request merged**